// ToiChat Buffer Protocol 
// Messages
//
// Created on: 02/01/2016
// Author: Toi-Group
//
package ToiChatProtocol;

message ToiChatMessage {
    oneof messageType {
        DnsMessage dnsMessage = 1;
        ChatMessage chatMessage = 2;
    }
}

message DnsMessage {
    // Gather message about client sending a new message
    //
    required string command = 1;
    required string clientName = 2; // Hostname/CallSign
    required string clientId = 3; // IPaddress
    required string lastUpdate = 4;
    optional string description = 5 [default = "NONE"]; // Misc Info.
    message DNSClients {
        required string clientName = 6; // Hostname/CallSign
        required string clientId = 7; // IPaddress
        required string lastUpdate = 8;
        optional string description = 9 [default = "NONE"]; // Misc Info.
    }
    repeated DNSClients clients = 6;
}

message ChatMessage {
    required string id = 1;
}

